code/core/src/manager-api/modules/url.ts

```diff
@@ -183,6 +225,51 @@ export const init: ModuleFn<SubAPI, SubState> = (moduleArgs) => {
   };
 
   const api: SubAPI = {
+    getStoryHrefs(storyId, options = {}) {
+      const { id: currentStoryId, refId: currentRefId } = fullAPI.getCurrentStoryData() ?? {};
+      const isCurrentStory = storyId === currentStoryId && options.refId === currentRefId;
+
+      const { customQueryParams, location, refs } = store.getState();
+      const {
+        base,
+        inheritArgs = isCurrentStory,
+        inheritGlobals = true,
+        queryParams = {},
+        refId,
+        viewMode = 'story',
+      } = options;
+
+      if (refId && !refs[refId]) {
+        throw new Error(`Invalid refId: ${refId}`);
+      }
+
+      const originAddress = global.window.location.origin + location.pathname;
+      const networkAddress = global.STORYBOOK_NETWORK_ADDRESS ?? originAddress;
+      const managerBase =
+        base === 'origin' ? originAddress : base === 'network' ? networkAddress : location.pathname;
+      const previewBase = refId
+        ? refs[refId].url + '/iframe.html'
+        : global.PREVIEW_URL || `${managerBase}iframe.html`;
+
+      const refParam = refId ? `&refId=${encodeURIComponent(refId)}` : '';
+      const { args = '', globals = '', ...otherParams } = queryParams;
+      let argsParam = inheritArgs
+        ? mergeSerializedParams(customQueryParams?.args ?? '', args)
+        : args;
+      let globalsParam = inheritGlobals
+        ? mergeSerializedParams(customQueryParams?.globals ?? '', globals)
+        : globals;
+      let customParams = stringify(otherParams);
```

Comment:
We do have rather inconsistent usage of this stringify util. In some code paths, we ensure it supports nesting, and in others, we don't.

I believe `nesting: true` would be safer. This is what router utils use:
```ts
  return stringify(encodeSpecialValues(object), {
    delimiter: ';', // we don't actually create multiple query params
    nesting: true,
    nestingSyntax: 'js', // encode objects using dot notation: obj.key=val
  })
```

Ideally we'd call this directly: https://github.com/storybookjs/storybook/blob/next/code/core/src/router/utils.ts#L186-L214; if not possible we should ensure logics are identical.

Reply:
We should be able to use it, I think.